packages/p2p/src/components/modal-manager/modals/filter-modal/filter-modal.jsx

```diff
@@ -157,6 +160,11 @@ const FilterModal = () => {
         hideModal();
     };
 
+    const onToggle = () => {
+        buy_sell_store.setShouldUseClientLimits(!buy_sell_store.should_use_client_limits);
+        if (!has_made_changes) setHasMadeChanges(is_matching !== buy_sell_store.should_use_client_limits);
```

Comment:
added check as a safe guard as if the user did make changes, and toggles the matching ads toggle, this could affect the has_made_changes value